imap_processing/cdf/cdf_attribute_manager.py

```diff
@@ -0,0 +1,207 @@
+"""Class for creating and managing CDF attrs.
+
+Developed based of HermesDataSchema from HERMES-SOC/hermes_core.
+"""
+
+from __future__ import annotations
+
+from pathlib import Path
+
+import yaml
+
+DEFAULT_GLOBAL_CDF_ATTRS_FILE = "imap_default_global_cdf_attrs.yaml"
+DEFAULT_GLOBAL_CDF_ATTRS_SCHEMA_FILE = "default_global_cdf_attrs_schema.yaml"
+DEFAULT_VARIABLE_CDF_ATTRS_SCHEMA_FILE = "default_variable_cdf_attrs_schema.yaml"
+
+
+class CdfAttributeManager:
+    """
+    Class for creating and managing CDF attributes based out of yaml files.
+
+    This class is based on the HERMES SWxSOC project for managing CDF attributes, but
+    is intended to be a flexible and very lightweight way of managing CDF attribute
+    creation and validation.
+
+    To use, you can load one or many global and variable attribute files:
+
+    ```
+    cdf_attr_manager = CdfAttributeManager(data_dir)
+    cdf_attr_manager.load_global_attributes("global_attrs.yaml")
+    cdf_attr_manager.load_global_attrs("instrument_global_attrs.yaml")
+    cdf_attr_manager.load_variable_attrs("variable_attrs.yaml")
+    ```
+
+    You can then get the global and variable attributes:
+
+    ```
+    # Instrument ID is optional for refining the attributes used from the file
+    global_attrs = cdf_attr_manager.get_global_attributes(instrument_id)
+    variable_attrs = cdf_attr_manager.get_variable_attributes(variable_name)
+    ```
+
+    The variable and global attributes are validated against the schemas upon calling
+    `get_global_attributes` and `get_variable_attributes`.
+
+    Attributes
+    ----------
+    source_dir: Path
+        The directory containing the schema and variable files - nominally config/
+    """
+
+    def __init__(self, data_dir: Path):
+        """
+        Initialize the CdfAttributeManager and read schemas from data_dir.
+
+        Parameters
+        ----------
+        data_dir: Path
```

Comment:
numpydoc should have a space before the colon as well.
```suggestion
        data_dir : Path
```